comp/core/workloadmeta/collectors/internal/remote/processcollector/process_collector_test.go

```diff
@@ -254,11 +270,12 @@ func TestCollection(t *testing.T) {
 
 			time.Sleep(time.Second)
 
+			ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
can I suggested to use `context.Background()` instead of `context.TODO()`

If I remember they are providing similar functionality, however TODO() give the impression this need to be change in the futur. 

```suggestion
			ctx, cancel := context.WithCancel(context.Background())
```

also imo, it doesn't seems that we need a context.WithCancel() at this point because you create another one in the `newMockServer()` function

Reply:
Agree about using `context.Background()`

>  at this point because you create another one in the newMockServer() function

You are right, but the created context is used to start the collector [here](https://github.com/DataDog/datadog-agent/blob/79cc32a50cc5e11f788c031cd2f9ce921072b3ef/comp/core/workloadmeta/collectors/internal/remote/processcollector/process_collector_test.go#L310). I thought I'd use the same context so that if the collector's context is cancelled, the streaming server is stopped automatically.

Reply:
Ok didn't catch that you have just moved the ctx creation earlier in the test. 👍 

in that case since the `cancel` function is already all later in the test. maybe you don't need to create another `context.WithCancel()` in the newMockServer() and so no `stop()` function since the cancel will cascade the cancel on your internal `mockserver.ctx`.

Reply:
Yes this can work too, but in that case we will not be able to stop the streaming server without stopping the collector. 

This is something we might want to test.

Reply:
I feel like the `cancel()` function will always be executed before the `mockServer.stop()` function because we are using a `defer` to call the  `mockServer.stop()` and not for the `cancel()` function

So I suspect that the cancel cascading will always appends before that `mockServer.stop()` is executed.

Reply:
yes correct
But this is because here we don't need to stop the streaming server before the tests finish.
Calling defer server.stop() is just a safety call to ensure the server is stopped at the end, even if the collector didn't stop.
Basically, with what I have now, there is a dependency as follows:
If the collector stops, the streaming server stops.
We can eliminate this dependency by not cascading the contexts of the server and the collector:

```
ctx := context.Background()

server := newMockServer(ctx)
defer server.stop()

collectorCtx, collectorCancel := context.WithCancel(ctx)

collector.start(collectorCtx)
defer collectorCancel()
```